core/commonMain/src/ArgParser.kt

```diff
@@ -625,6 +626,13 @@ open class ArgParser(
                     }
                 } else {
                     usedSubcommand = subcommands[arg]
+                    if (strictSubcommandOptionsOrder) {
+                        return usedSubcommand!!.run {
```

Comment:
This return breaks setting default values for options and arguments. Default values are set in the next block.